components/brave_wallet/browser/json_rpc_service.cc

```diff
@@ -806,15 +805,16 @@ void JsonRpcService::OnGetBlockNumber(GetBlockNumberCallback callback,
   std::move(callback).Run(block_number, mojom::ProviderError::kSuccess, "");
 }
 
-void JsonRpcService::GetFeeHistory(GetFeeHistoryCallback callback) {
+void JsonRpcService::GetFeeHistory(const std::string& chain_id,
+                                   GetFeeHistoryCallback callback) {
   auto internal_callback =
       base::BindOnce(&JsonRpcService::OnGetFeeHistory,
                      weak_ptr_factory_.GetWeakPtr(), std::move(callback));
 
   RequestInternal(eth::eth_feeHistory("0x28",  // blockCount = 40
                                       kEthereumBlockTagLatest,
                                       std::vector<double>{20, 50, 80}),
-                  true, network_urls_[mojom::CoinType::ETH],
+                  true, GetNetworkURL(prefs_, chain_id, mojom::CoinType::ETH),
```

Comment:
We'll hit DCHECK in RequestInternal if it's invalid url because caller might give us an invalid chain_id that we couldn't get network URL from it.
Can we maybe error out (calling callbacks with error) in RequestInternal when this happens?

Reply:
fixed in 210bbec4829bea9ca6ae39e73d2bbd6bb233ad9e